src/test.js

```diff
@@ -733,17 +733,11 @@ function internalStart( test ) {
 }
 
 function numberOfTests( module ) {
-	let count = module.tests.length,
-		modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
-	while ( modules.length ) {
-		let nextModule =  modules.shift();
-		count += nextModule.tests.length;
-		modules.push( ...nextModule.childModules );
-	}
-
-	return count;
+	return module.childModules.reduce( function( p, childModule ) {
+	    return p + numberOfTests( childModule );
```

Comment:
if we change the signature of numberOfTests to receive a offset param, we can take some advantage from Tail Call Optimization, consuming less memory in a long list of modules.

That would be like: 

```js
function numberOfTests( module, offset = 0 ) {

	return module.childModules.reduce( function( p, childModule ) {
		return numberOfTests( childModule, p );
	...
```

too bad the args order are the inverse, otherwise it would be even easier.

At least this is also better using arrow fns:

```js
function numberOfTests( module, offset = 0 ) {
	return module.childModules.reduce( ( p, childModule ) => numberOfTests( childModule, p ) );
```